nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/main/java/org/nuxeo/elasticsearch/audit/ESAuditMigrationWork.java

```diff
@@ -80,9 +81,16 @@ public void work() {
             int pageIdx = 1;
 
             while (nbEntriesMigrated < nbEntriesToMigrate) {
-                @SuppressWarnings("unchecked")
-                List<LogEntry> entries = (List<LogEntry>) sourceBackend.nativeQuery(
-                        "from LogEntry log order by log.id asc", pageIdx, batchSize);
+                boolean txStarted = TransactionHelper.startTransaction();
+                List<LogEntry> entries;
+                if (txStarted) {
+                    // @SuppressWarnings("unchecked")
+                    entries = (List<LogEntry>) sourceBackend.nativeQuery(
+            	                         "from LogEntry log order by log.id asc", pageIdx, batchSize);
+                    TransactionHelper.commitOrRollbackTransaction();
```

Comment:
The proper idiom to run something in a transaction with correct error checking is:

    List<LogEntry> entries = TransactionHelper.runInTransaction(() -> (List<LogEntry>) sourceBackend.nativeQuery(
                         "from LogEntry log order by log.id asc", pageIdx, batchSize));